core/dbt/task/base.py

```diff
@@ -309,7 +310,7 @@ def skip_result(self, node, message):
 
     def compile_and_execute(self, manifest, ctx):
         result = None
-        with self.adapter.connection_for(self.node):
+        with self.adapter.connection_for(self.node) if get_flags().INTROSPECT else nullcontext():
```

Comment:
If a user set `DBT_INTROSPECT` to False, but we are not running compile/preview command, `get_flags().INTROSPECT` will be `True` since it is using default value right?